### ISSUE TITLE
BSOD on Windows 10 with Reparent function

### ISSUE THREAD

**Issue by skeetor** [wxWidgets/wxWidgets#22794]
I know that a BSOD is not strictly a wxWidgets problem itself, nevertheless I thought I report here just in case.

I was using the latest build from git with Visual Studio 2019. I think that is not the main issue though, because the BSOD happens in a Win32 function which is called in a base function.

```
bool wxNonOwnedWindow::Reparent(wxWindowBase* newParent)
{
...
    ::SetWindowLongPtr(GetHwnd(), GWLP_HWNDPARENT, (LONG_PTR)hwndOwner);
...
}
```

I tested it on two different machines and it on both it crashed the machine.

```
void MyFrame::bsod()
{
    wxSplitterWindow *sp = new wxSplitterWindow(this);
    this->Reparent(sp);
}
```

**Comment by vadz** [comment]
Nothing a program does should result in a BSOD so it's definitely a Windows bug and I advise reporting it to Microsoft.

This being said, it appears that you're reparenting a window under its own child, which probably should be disallowed in the first place but, again, shouldn't result in a crash, of course.

**Comment by skeetor** [comment]
I know and I already fixed that bug. I just investigated it, because I wanted to know where it comes from, so I found that location. Do you know how I can report this? A BSOD form user space is pretty serious in my opinion.

**Comment by vadz** [comment]
No, sorry, I don't know what's the current of reporting such bugs.
